src/util/notebookLegacy.ts

```diff
@@ -1,30 +1,31 @@
+import { toVscodeEditor } from "@cursorless/vscode-common";
 import { range } from "lodash";
 import * as semver from "semver";
-import {
-  commands,
-  NotebookDocument,
-  TextDocument,
-  TextEditor,
-  version,
-} from "vscode";
-import { getNotebookFromCellDocument } from "./notebook";
+import { commands, NotebookDocument, TextDocument, version } from "vscode";
+import { VscodeEditableTextEditorImpl } from "../ide/vscode/VscodeEditableTextEditorImpl";
 import { getCellIndex } from "../libs/vscode-common/notebook";
-import { getActiveTextEditor } from "../ide/vscode/activeTextEditor";
+import { getNotebookFromCellDocument } from "./notebook";
 
 export function isVscodeLegacyNotebookVersion() {
   return semver.lt(version, "1.68.0");
 }
 
-export async function focusNotebookCellLegacy(editor: TextEditor) {
-  const activeTextEditor = getActiveTextEditor();
+export async function focusNotebookCellLegacy(
+  editor: VscodeEditableTextEditorImpl,
+) {
+  const activeTextEditor = editor.ide.activeTextEditor;
```

Comment:
Notice that I now get this from the editor instead of using the `ide` from ambient context

Reply:
Am not a big fan of this solution. I guess it's only for vscode notebooks but still feels like a anti pattern.

Reply:
Why is this an anti-pattern?

Reply:
ide is a singleton and responsible for creating editors. Putting that as a member of an editor feels really weird.

Reply:
Ok but today ide is a global. Exactly everywhere else in the code we fetch from the same resource except here that why I think this is a weird pattern. I don't feel strongly enough about it but I still think it's a bit weird.